api/src/main/java/net/starype/quiz/api/game/AnswerEligibility.java

```diff
@@ -0,0 +1,21 @@
+package net.starype.quiz.api.game;
+
+import net.starype.quiz.api.game.player.UUIDHolder;
+
+import java.util.Collection;
+
+public interface AnswerEligibility {
+
+    boolean isEligible(UUIDHolder player);
+    default boolean existsEligible(Collection<? extends UUIDHolder> players) {
+        return players.stream().anyMatch(this::isEligible);
+    }
+
+    default AnswerEligibility or(AnswerEligibility other) {
+        return (player) -> this.isEligible(player) || other.isEligible(player);
+    }
+
+    default AnswerEligibility and(AnswerEligibility other) {
+        return (player) -> this.isEligible(player) && other.isEligible(player);
+    }

```

Comment:
The name of this class suggest that it perform a check if the answer is correctly formatted. So it doesn't make sense that it takes UUIDHolder as input. You could rename this ```EntityEligibility```